artemis-server/src/test/java/org/apache/activemq/artemis/tests/util/ActiveMQTestBase.java

```diff
@@ -619,34 +623,107 @@ public static void checkWeakReferences(final WeakReference<?>... references) {
    }
 
    public static String threadDump(final String msg) {
-      StringWriter str = new StringWriter();
-      PrintWriter out = new PrintWriter(str);
 
-      Map<Thread, StackTraceElement[]> stackTrace = Thread.getAllStackTraces();
+      try (
```

Comment:
Hmmmm... now that you have something complete like this.. we need to think of a way to reuse this logic into ActiveMQServerImpl::threadDump

Reply:
Perhaps add a class into util: String ThreadDumpUtil.threadDump()

And call this on both places?


Do you want to take care of this, or should I do it?

Reply:
Thanks for review. I will do it.